src/table.ts

```diff
@@ -1872,6 +1887,98 @@ class Table extends common.ServiceObject {
         ? optionsOrCallback
         : (cb as InsertRowsCallback);
 
+    this._insertAndCreateTable(rows, options).then(
+      resp => callback(null, resp),
+      err => callback(err, null)
+    );
+  }
+
+  /**
+   * Insert rows with retries, but will create the table if not exists.
+   *
+   * @param {RowMetadata | RowMetadata[]} rows
+   * @param {InsertRowsOptions} options
+   * @returns {Promise<bigquery.ITableDataInsertAllResponse | bigquery.ITable>}
+   * @private
+   */
+  private async _insertAndCreateTable(
+    rows: RowMetadata | RowMetadata[],
+    options: InsertRowsOptions
+  ): Promise<bigquery.ITableDataInsertAllResponse | bigquery.ITable> {
+    const {schema} = options;
+    const delay = 60000;
+
+    try {
+      return await this._insertWithRetry(rows, options);
+    } catch (err) {
+      if ((err as common.ApiError).code !== 404 || !schema) {
+        throw err;
+      }
+    }
+
+    try {
+      await this.create({schema});
+    } catch (err) {
+      if ((err as common.ApiError).code !== 409) {
+        throw err;
+      }
+    }
+
+    // table creation after failed access is subject to failure caching and
+    // eventual consistency, see:
+    // https://github.com/googleapis/google-cloud-python/issues/4553#issuecomment-350110292
+    await new Promise(resolve => setTimeout(resolve, delay));
+    return this._insertAndCreateTable(rows, options);
+  }
+
+  /**
+   * This method will attempt to insert rows while retrying any partial failures
+   * that occur along the way. Because partial insert failures are returned
+   * differently, we can't depend on our usual retry strategy.
+   *
+   * @private
+   *
+   * @param {RowMetadata|RowMetadata[]} rows The rows to insert.
+   * @param {InsertRowsOptions} options Insert options.
+   * @returns {Promise<bigquery.ITableDataInsertAllResponse>}
+   */
+  private async _insertWithRetry(
+    rows: RowMetadata | RowMetadata[],
+    options: InsertRowsOptions
+  ): Promise<bigquery.ITableDataInsertAllResponse> {
+    const maxAttempts = options.maxAttempts || 3;
```

Comment:
Nice catch!

Reply:
currently `maxAttempts` includes the _first_ request, regardless of failure; so if a user wanted to try once, then retry up to 3 more times, `maxAttempts` would have to be 4. A value of `0` means don't even try, which is probably not what we want.

Additionally, this retry is specifically for _partial row insertion failures_ and not any other failure scenarios.

I've changed the option to `partialRetries`, modeled off of [`node-retry`](https://github.com/tim-kos/node-retry#retryoperationoptions and scoped to the "partial" concept, to avoid confusion.